xapian-letor/tests/api_letor.cc

```diff
@@ -179,7 +179,8 @@ DEFINE_TESTCASE(listnet_ranker, generated)
     TEST_EXCEPTION(Xapian::FileNotFoundError,
 		   ranker.score(qrel, "", "ListNet_Ranker",
 				"scorer_output.txt", 10));
-    ranker.score(query, qrel, "ListNet_Ranker", "scorer_output.txt", 10);
+    ranker.score(query, qrel, "ListNet_Ranker", "ndcg_output.txt", 10);
+    ranker.score(query, qrel, "ListNet_Ranker", "err_output.txt", 10, "ERRScore");
```

Comment:
These tests don't tell us whether `ERRScore` is implemented properly, just whether they throw an exception or not. You should be able to test scorers directly by constructing `FeatureVector`s and driving their `score()` method, rather than (as this does) driving the ranker's `score()` method.

Reply:
I have added a test for the err_scorer